frontend/src/features/admin-form/responses/ResponsesPage/storage/worker/decryption.worker.ts

```diff
@@ -49,48 +57,121 @@ function verifySignature(
+      const decryptedObject = formsgSdk.crypto.decrypt(secretKey, {
+        encryptedContent,
+        verifiedContent,
+        version,
+      })
+      if (!decryptedObject) {
+        const errorMessage = `Invalid decryption for storage mode response`
+        console.error(errorMessage)
+        return {
+          errorMessage,
+          isSubmissionDecryptionSuccessful: false,
+        }
+      }
+      decryptedResponses = processDecryptedContent(decryptedObject)
+      break
+    }
+    case SubmissionType.Multirespondent: {
+      const decryptedObject = formsgSdk.cryptoV3.decrypt(secretKey, {
+        encryptedSubmissionSecretKey:
+          submissionData.encryptedSubmissionSecretKey,
+        encryptedContent,
+        verifiedContent,
+        version,
+      })
+      if (!decryptedObject) {
+        const errorMessage = `Invalid decryption for multirespondent response`
+        console.error(errorMessage)
+        return {
+          errorMessage,
+          isSubmissionDecryptionSuccessful: false,
+        }
+      }
+      mrfSubmissionSecretKey = decryptedObject.submissionSecretKey
+      decryptedResponses = await processDecryptedContentV3(
+        submissionData.form_fields,
+        decryptedObject,
+      )
+      break
+    }
+    default: {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const _: never = submissionData
+      const errorMessage = `Invalid submission type encountered.`
+      console.error(errorMessage)
+      return {
+        errorMessage,
+        isSubmissionDecryptionSuccessful: false,
+      }
+    }
+  }
+  return {
+    decryptedResponses,
+    mrfSubmissionSecretKey,
+    isSubmissionDecryptionSuccessful: true,
+  }
+}
+
 /**
  * Decrypts given data into a {@type CsvRecord} and posts the result back to the
  * main thread.
  * @param data The data to decrypt into a csvRecord.
  */
 async function decryptIntoCsv(
   data: LineData,
-  isFasterDownloadsEnabled: boolean,
+  isFasterDownloadsEnabled: boolean = false,
 ): Promise<MaterializedCsvRecord> {
   // This needs to be dynamically imported due to sharing code between main app and worker code.
   // Fixes issue raised at https://stackoverflow.com/questions/66472945/referenceerror-refreshreg-is-not-defined
   // Something to do with babel-loader.
-
-  // TODO: May be removed when we move to Webpack 5, where web workers are now first class citizens?
-  const { processDecryptedContent, processDecryptedContentV3 } = await import(
-    '../utils/processDecryptedContent'
-  )
-  const { downloadAndDecryptAttachmentsAsZip } = await import(
-    '../utils/downloadAndDecryptAttachment'
-  )
-
-  const { SubmissionStreamDto, SubmissionType } = await import('~shared/types')
-
   const { line, secretKey, downloadAttachments, formId, hostOrigin } = data
-
-  let csvRecord: CsvRecord
   const attachmentDownloadUrls: AttachmentsDownloadMap = new Map()
   let downloadBlob: Blob
+  let csvRecord: CsvRecord
 
```

Comment:
nit(review): is this ordering change necessary?

Reply:
removed the ordering change